File: gen_scripts/ruby/f40.loop.rb
#!/usr/bin/env ruby
# pure spin loop using continous loop construct
#   break is used to exit loop, next to skip loop
loop do
  print "Enter your name (quit to Exit): " # print prompt
  STDOUT.flush                            # flush buffer to show prompt
  answer = gets.chomp                     # get string input

  if answer =~ /^[\s\t]*$/
    next                                  # skip loop if not answer entered
  end

  if answer.chomp == "quit"
    break                                 # exit loop if user wants to quit
  end

  puts "Hello #{answer}!"                 # print result as not exiting
end

File: gen_scripts/perl/f40.loop.pl
#!/usr/bin/env perl -w
# spin loop construction with last to exiting
while (1) {
  print "Enter your name (quit to Exit): "; # prompt user
  chomp($answer = <>);                      # git input and trim newline
  next if $answer =~ /^[\t\s]*$/;           # skip loop if not answer entered
  last if $answer eq "quit";                # exit loop if user wants to quit
  print "Hello $answer!\n";                 # output results as not exiting
}
